src/test/groovy/org/gradle/android/SimpleAndroidApp.groovy

```diff
@@ -0,0 +1,192 @@
+package org.gradle.android
+
+class SimpleAndroidApp {
+    final File projectDir
+    private final File cacheDir
+    final String androidVersion
+    private final boolean reverseApply
+
+    SimpleAndroidApp(File projectDir, File cacheDir, String androidVersion, boolean reverseApply = false) {
+        this.reverseApply = reverseApply
+        this.projectDir = projectDir
+        this.cacheDir = cacheDir
+        this.androidVersion = androidVersion
+    }
+
+    def writeProject() {
+        def app = 'app'
+        def appPackage = 'org.gradle.android.example.app'
+        def appActivity = 'AppActivity'
+
+        def library = 'library'
+        def libPackage = 'org.gradle.android.example.library'
+        def libraryActivity = 'LibraryActivity'
+
+        file("settings.gradle") << """
+                buildCache {
+                    local(DirectoryBuildCache) {
+                        directory = "${cacheDir.absolutePath.replace(File.separatorChar, '/' as char)}"
+                    }
+                }
+            """
+
+        file("build.gradle") << """
+                buildscript {
+                    repositories {
+                        google()
+                        jcenter()
+                        mavenLocal()
```

Comment:
I think you are using `mavenLocal()` to publish the plugin and consume it in the test. I think it would be better to use a custom file repository instead, so it would be contained within the project root.

Reply:
Hm, that'd make sense. Care to craft a PR?

Reply:
Perhaps tomorrow...